sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Common/MetricsTest.cs

```diff
@@ -581,8 +582,10 @@ private static MetricsBase MetricsFor(AnalyzerLanguage language, string text)
                 throw new ArgumentException("Supplied language is not C# neither VB.Net", nameof(language));
             }
 
+            (var syntaxTree, var semanticModel) = TestHelper.Compile(text);
+
             return language == AnalyzerLanguage.CSharp
-                ? (MetricsBase)new SonarAnalyzer.Metrics.CSharp.Metrics(CSharpSyntaxTree.ParseText(text))
+                ? (MetricsBase)new SonarAnalyzer.Metrics.CSharp.Metrics(syntaxTree, semanticModel)
                 : new SonarAnalyzer.Common.VisualBasic.Metrics(VisualBasicSyntaxTree.ParseText(text));
```

Comment:
Do we need a separate issue to fix the metrics for VB too?

Reply:
It's not implemented for VB.